README.md

```diff
@@ -821,3 +821,24 @@ Returns the number of bins according to [Sturges’ formula](https://en.wikipedi
 You may also implement your own threshold generator taking three arguments: the array of input [*values*](#bin_value) derived from the data, and the [observable domain](#bin_domain) represented as *min* and *max*. The generator may then return either the array of numeric thresholds or the *count* of bins; in the latter case the domain is divided uniformly into approximately *count* bins; see [ticks](#ticks).
 
 For instance, when binning date values, you might want to use the ticks from a time scale ([Example](https://observablehq.com/@d3/d3-bin-time-thresholds)).
+
+
+### Maps and Sets
+
+<a name="InternMap" href="#InternMap">#</a> new d3.<b>InternMap</b>([<i>iterable</i>][,<i>key</i>]) · [Source](https://github.com/mbostock/internmap/blob/main/src/index.js), [Examples](https://observablehq.com/d/d4c5f6ad343866b9)
+
+<a name="InternSet" href="#InternSet">#</a> new d3.<b>InternSet</b>([<i>iterable</i>][,<i>key</i>]) · [Source](https://github.com/mbostock/internmap/blob/main/src/index.js), [Examples](https://observablehq.com/d/d4c5f6ad343866b9)
+
+The InternMap and InternSet classes are an extension of the native JavaScript Map and Set classes, allowing to use Dates (or other objects) as keys, bypassing the [SameValueZero algorithm](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) which determines key equality. See [this repository](https://github.com/mbostock/internmap) for details. d3.group, d3.rollup and d3.index use an InternMap rather than a native Map.
+
+These two classes are exported for convenience. Note that a Map can be upgraded to an InternMap simply by calling d3.InternMap(map):
+```js
+const g = new d3.InternMap(
+  d3.group(
+    [[["foo", "bar"], 1], [["foo", "baz"], 2], [["goo", "bee"], 3]],
+    d => d[0]
+  ),
+  JSON.stringify
+);
```

Comment:
I wouldn’t recommend this approach: the interning would only apply *after* grouping, so any duplicate keys in the original entries would still be considered separate groups, and then the last of the duplicate groups would “win” in the returned InternMap rather than being combined. I think for simplicity’s sake we should just drop this note and example.